src/main/com/intellij/lang/jsgraphql/GraphQLFileType.java

```diff
@@ -20,54 +20,55 @@
 import javax.swing.*;
 
 public class GraphQLFileType extends LanguageFileType {
-  public static final GraphQLFileType INSTANCE = new GraphQLFileType();
+    public static final GraphQLFileType INSTANCE = new GraphQLFileType();
 
-  private GraphQLFileType() {
-    super(GraphQLLanguage.INSTANCE);
-  }
+    private GraphQLFileType() {
+        super(GraphQLLanguage.INSTANCE);
+    }
 
-  @NotNull
-  @Override
-  public String getName() {
-    return "GraphQL";
-  }
+    @NotNull
+    @Override
+    public String getName() {
+        return "GraphQL";
+    }
 
-  @NotNull
-  @Override
-  public String getDescription() {
-    return getName();
-  }
+    @NotNull
+    @Override
+    public String getDescription() {
+        return getName();
+    }
 
-  @NotNull
-  @Override
-  public String getDefaultExtension() {
-    return "graphql";
-  }
+    @NotNull
+    @Override
+    public String getDefaultExtension() {
+        return "graphql";
+    }
 
-  @Nullable
-  @Override
-  public Icon getIcon() {
-    return GraphQLIcons.FILE;
-  }
+    @Nullable
+    @Override
+    public Icon getIcon() {
+        return GraphQLIcons.FILE;
+    }
 
-  /**
-   * Scratch virtual files don't return their actual file type, so we need to find the PsiFile to determine
-   * whether it's a GraphQL scratch file
-   * @return true if the scratch file contains a GraphQL PsiFile
-   */
-  public static boolean isGraphQLScratchFile(Project project, VirtualFile file) {
-    if(file.getFileType() instanceof ScratchFileType) {
-      final PsiManager psiManager = PsiManager.getInstance(project);
-      try {
-        final PsiFile psiFile = psiManager.findFile(file);
-        if (psiFile != null && psiFile.getFileType() == GraphQLFileType.INSTANCE) {
-          return true;
+    /**
+     * Scratch virtual files don't return their actual file type, so we need to find the PsiFile to determine
+     * whether it's a GraphQL scratch file
+     *
+     * @return true if the scratch file contains a GraphQL PsiFile
+     */
+    public static boolean isGraphQLScratchFile(Project project, VirtualFile file) {
+        if (ScratchUtil.isScratch(file)) {
+            final PsiManager psiManager = PsiManager.getInstance(project);
+            try {
+                final PsiFile psiFile = psiManager.findFile(file);
+                if (psiFile != null && psiFile.getFileType() == GraphQLFileType.INSTANCE) {
+                    return true;
+                }
+            } catch (ProcessCanceledException e) {
+                // can be thrown from psiManager.findFile
```

Comment:
should be rethrown, I believe

Reply:
It shouldn't be caught at all here, but I believe there is a reason why it's here. I'll check it.